### ISSUE TITLE
Clang-tidy incorrect misc-const-correctness

### ISSUE THREAD

**Issue by jeremy-rifkin** [llvm/llvm-project#64113]
```cpp
#include <sstream>
#include <type_traits>

auto foo(int t) {
  std::ostringstream oss;
  oss<<t;
  return std::move(oss).str();
}

template<typename T>
std::string stringify(const T& t) {
    if constexpr(std::is_arithmetic<T>::value) {
        std::ostringstream oss; // <------ line 13
        oss<<t;
        return std::move(oss).str();
    } else {
        return "";
    }
}
```

```cpp
[<source>:13:9: warning: variable 'oss' of type 'std::ostringstream' (aka 'basic_ostringstream<char>') can be declared 'const' [misc-const-correctness]](javascript:;)
   13 |         std::ostringstream oss;
      |         ^
      |                            const
```

https://godbolt.org/z/zf6GGWTq1

This is an incorrect diagnostic, the code can never be valid if `oss` is `const`.

**Comment by firewave** [comment]
Duplicate of #57297.

**Comment by jeremy-rifkin** [comment]
Thanks!
